lib/matplotlib/axes/_base.py

```diff
@@ -2227,41 +2227,58 @@ def margins(self, *margins, x=None, y=None, tight=True):
         """
         Set or retrieve autoscaling margins.
 
-        signatures::
+        The padding added to each limit of the axes is the *margin*
+        times the data interval. All input parameters must be floats
+        within the range [0, 1]. Passing both positional and keyword
+        arguments is invalid and will raise a TypeError. If no
+        arguments (positional or otherwise) are provided, the current
+        margins will remain in place and simply be returned.
 
-            margins()
-
-        returns xmargin, ymargin
-
-        ::
-
-            margins(margin)
-
-            margins(xmargin, ymargin)
+        Specifying any margin changes only the autoscaling; for example,
+        if *xmargin* is not None, then *xmargin* times the X data
+        interval will be added to each end of that interval before
+        it is used in autoscaling.
 
-            margins(x=xmargin, y=ymargin)
+        Parameters
+        ----------
+        args : float, optional
+            If a single positional argument is provided, it specifies
+            both margins of the x-axis and y-axis limits. If two
+            positioanl arguments are provided, they will be interpreted
```

Comment:
There is a small typo here (positioanl), that I am happy to fix by pushing to your branch.

Reply:
Thanks for catching that! Fixed in latest (force) push.